crates/bdk/src/wallet/coin_selection.rs

```diff
@@ -531,10 +517,19 @@ impl BranchAndBoundCoinSelection {
         optional_utxos.sort_unstable_by_key(|a| a.effective_value);
         optional_utxos.reverse();
 
+        let mut curr_value = required_utxos
```

Comment:
I don't think it's worth the hassle to refactor the cs module, since it's going away soon. Also see https://github.com/bitcoindevkit/bdk/issues/1127#issuecomment-1784313256